### ISSUE TITLE
win_disk_facts: Should expose raw values

### ISSUE THREAD

**Issue by dagwieers** [ansible/ansible#35329]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
win_disk_facts

##### ANSIBLE VERSION
v2.5

##### SUMMARY
Currently there's no way to get the disk size in a format you can manipulate. For instance, the disk ought to be a 240GB disk. However the value is returned as:
```yaml
                "size": "223.57088470459GiB", 
```
But since it is a string, we cannot convert to a float and recalculate in GB. Which we could be doing like this:
```yaml
  - name: Isolate disk
    set_fact:
      disk: '{{ ansible_facts.disks|selectattr("system_disk")|first }}'

  - name: Set disk size
    set_fact:
      disk_size: '{{ (disk.size|float*1024*1024/1000/1000)|round|int }}GB'
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/windows/win_disk_facts.ps1](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_disk_facts.ps1)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @jborean93 @jhawkesworth @marqelme @nitzmahone
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jborean93** [comment]
We talked about this in the review, the thinking was to add a new parameter where you could specify the units of these values if you wished to override the default.

**Comment by dagwieers** [comment]
Hmm, I think it's best to return the raw values and leave the conversion up to the user.
He can then decide for himself in what unit he wants the values to be.

**Comment by jborean93** [comment]
I probably am more on the side that the default value should be raw bytes but @marqelme and others internally thought to keep them as human readable values instead and this would be undoing his work there. I think we should implement the unit selector into this PR (or we not touch the size in this one) so that everybody is happy.

**Comment by dagwieers** [comment]
I am sorry I missed that discussion, however I think it is a big mistake not to do the right thing now IMO.
If we don't do it now, we will be bound by leaving it as-is forever.

If we haven't learned anything from the default Unix facts output for date/time information, we haven't learned anything at all.
```yaml
        "ansible_date_time": {
            "date": "2018-01-25", 
            "day": "25", 
            "epoch": "1516849910", 
            "hour": "04", 
            "iso8601": "2018-01-25T03:11:50Z", 
            "iso8601_basic": "20180125T041150437197", 
            "iso8601_basic_short": "20180125T041150", 
            "iso8601_micro": "2018-01-25T03:11:50.437272Z", 
            "minute": "11", 
            "month": "01", 
            "second": "50", 
            "time": "04:11:50", 
            "tz": "CET", 
            "tz_offset": "+0100", 
            "weekday": "donderdag", 
            "weekday_number": "4", 
            "weeknumber": "04", 
            "year": "2018"
        }, 
```
Size must be an integer value, and not a string, and it must contain the most detailed representation so we don't loose the possibility to use it for whatever purpose. There is no need to have a formatted string, the below works fine with a raw value:

```yaml
  vars:
    # Get first system disk
    disk: '{{ ansible_facts.disks|selectattr("system_disk")|first }}'

    # Show disk size in Gibibytes
    disksize_gib_human: '{{ disk.size|filesizeformat(True) }}'   # returns "223.6 GiB" (human readable)
    disksize_gib: '{{ (disk.size/1024|pow(3))|round|int }} GiB'  # returns "224 GiB" (value in GiB)

    # Show disk size in Gigabytes
    disksize_gb_human: '{{ disk.size|filesizeformat }}'        # returns "240.1 GB" (human readable)
    disksize_gb: '{{ (disk.size/1000|pow(3))|round|int }} GB'  # returns "240 GB" (value in GB)
```

We probably should add the above examples in the module documentation.

**Comment by jborean93** [comment]
I personally do think we should change it and represent it as raw bytes (it's what I advocated originally) but I am just wary of doing it without a wider discussion and agreement.

PS. donderdag is my new favourite translation for Thursday

**Comment by marqelme** [comment]
@jborean93 @dagwieers 
As I can see you have already changed it here?
https://github.com/ansible/ansible/pull/35326

Should I change or do anything now or is the task on your side?

**Comment by dagwieers** [comment]
@marqelme I made a PR for the namespace issue, and then for further testing locally I started adding the other changes as well. Mostly because that was the most convenient thing to do (I have no clue how to test a module with changes from 2 separate branches, I must confess :-/)

I don't mind separating the namespace/doc fixes from the size string/int changes.

Only after creating these later commits @jborean wanted to discuss this first with all existing stakeholders. So we are mostly waiting for you and possibly others to voice their opinion what we should do before setting this in stone with the v2.5 release.

*To be honest, I feel partly responsible for this because I hadn't noticed we were returning these size values as strings, or I didn't realize the implications until I started actively using your facts module in our CSV collect script, these issues popped up.*

**Comment by marqelme** [comment]
Ok sure, no worries.
But I am also not aware how to test this now.
Regarding the units, I am fine with the RAW format, there are pros and cons with both of the approaches.
But if the user should be able to use the output in order to convert it in any possible unit the RAW format is better way.

**Comment by dagwieers** [comment]
You can pull my branch (e.g. add a new remote, fetch and then checkout the branch), or simply copy&paste the raw file in Github and put it in your library directory.

**Comment by marqelme** [comment]
Ok and the I will create a new PR?

**Comment by marqelme** [comment]
@dagwieers @jborean93 
I have created pull request #35345 with the 2 changed files.

**Comment by dagwieers** [comment]
@marqelme Why another PR ?

**Comment by marqelme** [comment]
Then I didn't get what you mean.
My branch which was merged is already deleted.

**Comment by dagwieers** [comment]
You asked how to test this, so I gave you feedback on how you could test the module from my branch/PR.
There is no need to make your own branch and create a new PR.

**Comment by marqelme** [comment]
Then we talked past each other, sorry!
